### PR TITLE
fix: masking password property in logs

### PR DESCRIPTION
**Pull request by brunos-bq**
### Summary

Redacting password property when printed in logs

Related to #693 

### Additional Reviewers

@crystall-bitquill 
@sergiyv-improving

### CODE REVIEW

File: wrapper/src/main/java/software/amazon/jdbc/plugin/failover/ClusterAwareReaderFailoverHandler.java [hunk 1]
```diff
@@ -390,10 +391,11 @@ private ConnectionAttemptTask(final HostSpec newHost) {
      */
     @Override
     public ReaderFailoverResult call() {
+      Properties redactedProperties = PropertyUtils.redactedProperties(initialConnectionProps);
```
Comment (sergiyvamz) [inline wrapper/src/main/java/software/amazon/jdbc/plugin/failover/ClusterAwareReaderFailoverHandler.java:394]: maskedProperties() ?

Comment (sergiyvamz) [reply wrapper/src/main/java/software/amazon/jdbc/plugin/failover/ClusterAwareReaderFailoverHandler.java:394]: I'd relocate this line to inside LOGGER.fine()  lambda.

File: wrapper/src/main/java/software/amazon/jdbc/plugin/failover/ClusterAwareWriterFailoverHandler.java [hunk 2]
```diff
@@ -235,10 +236,11 @@ public ReconnectToWriterHandler(final HostSpec originalWriterHost) {
     }
 
     public WriterFailoverResult call() {
+      Properties redactedProperties = PropertyUtils.redactedProperties(initialConnectionProps);
```
Comment (sergiyvamz) [inline wrapper/src/main/java/software/amazon/jdbc/plugin/failover/ClusterAwareWriterFailoverHandler.java:239]: similar note

File: wrapper/src/main/java/software/amazon/jdbc/util/PropertyUtils.java [hunk 3]
```diff
@@ -126,6 +126,14 @@ public static void setPropertyOnTarget(
     return copy;
   }
 
+  public static @NonNull Properties redactedProperties(final Properties props) {
+    Properties redactedProperties = copyProperties(props);
+    if (redactedProperties.containsKey("password")) {
```
Comment (sergiyvamz) [inline wrapper/src/main/java/software/amazon/jdbc/util/PropertyUtils.java:131]: contains(PropertyDefinition.PASSWORD)

File: wrapper/src/main/java/software/amazon/jdbc/util/PropertyUtils.java [hunk 4]
```diff
@@ -126,6 +126,14 @@ public static void setPropertyOnTarget(
     return copy;
   }
 
+  public static @NonNull Properties redactedProperties(final Properties props) {
+    Properties redactedProperties = copyProperties(props);
```
Comment (crystall-bitquill) [inline wrapper/src/main/java/software/amazon/jdbc/util/PropertyUtils.java:130]:
```suggestion
    final Properties redactedProperties = copyProperties(props);
```

**Review by crystall-bitquill — APPROVED** [review]

File: wrapper/src/main/java/software/amazon/jdbc/util/PropertyUtils.java [hunk 5]
```diff
@@ -126,6 +127,14 @@ public static void setPropertyOnTarget(
     return copy;
   }
 
+  public static @NonNull Properties maskProperties(final Properties props) {
+    final Properties maskedProperties = copyProperties(props);
+    if (maskedProperties.contains(PropertyDefinition.PASSWORD)) {
+      maskedProperties.setProperty("password", "***");
```
Comment (sergiyvamz) [inline wrapper/src/main/java/software/amazon/jdbc/util/PropertyUtils.java:133]: maskedProperties.setProperty(PropertyDefinition.PASSWORD, "***");

**Review by sergiyvamz — APPROVED** [review]

**Pull request merged**